### PR TITLE
[terraform-users] add support for role expiration

### PR DESCRIPTION
**Pull request by maorfr**
part of https://issues.redhat.com/browse/APPSRE-4338

follows up on #2033

this PR adds support for role expiration to terraform-users, by filtering out roles that are beyond the expiration date.

recommendation: review commit by commit.

### CODE REVIEW

File: reconcile/utils/expiration.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+from ast import List
```
Comment (geoberle) [inline reconcile/utils/expiration.py]: do you mean `from typing import List`?

File: reconcile/utils/expiration.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    else:
+        exp_date = datetime.datetime \
+            .strptime(role, '%Y-%m-%d').date()
+        current_date = datetime.datetime.utcnow().date()
+        if current_date < exp_date:
+            return True
+        return False
+
+
+def filter(roles: List[dict], key: str = 'expirationDate') -> List[dict]:
```
Comment (geoberle) [inline reconcile/utils/expiration.py:32]: you can also use `list[dict]` in current python so you don't need the import at all

File: reconcile/utils/expiration.py (hunk 1 above)
Comment (maorfr) [reply reconcile/utils/expiration.py]: i did!

File: reconcile/utils/expiration.py (hunk 2 above)
Comment (maorfr) [reply reconcile/utils/expiration.py]: done! thank you!

**Review by patjlm — APPROVED** [review]
LGTM

**Pull request merged**